server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIndex.java

```diff
@@ -381,77 +398,78 @@ private void configureStickyFacets(IssueQuery query, SearchOptions options, Map<
+      if (options.getFacets().contains(ASSIGNEES)) {
         esSearch.addAggregation(createAssigneesFacet(query, filters, esQuery));
       }
       addAssignedToMeFacetIfNeeded(esSearch, options, query, filters, esQuery);
-      if (options.getFacets().contains(IssueFilterParameters.ACTION_PLANS)) {
+      if (options.getFacets().contains(ACTION_PLANS)) {
         esSearch.addAggregation(createActionPlansFacet(query, filters, esQuery));
       }
-      if (options.getFacets().contains(IssueFilterParameters.CREATED_AT)) {
+      if (options.getFacets().contains(CREATED_AT)) {
         esSearch.addAggregation(getCreatedAtFacet(query, filters, esQuery));
       }
     }
 
-    if (IssueFilterParameters.FACET_MODE_DEBT.equals(query.facetMode())) {
-      esSearch.addAggregation(DEBT_AGGREGATION);
+    if (hasQueryEffortFacet(query)) {
+      esSearch.addAggregation(EFFORT_AGGREGATION);
     }
   }
 
-  private StickyFacetBuilder newStickyFacetBuilder(IssueQuery query, Map<String, FilterBuilder> filters, QueryBuilder esQuery) {
-    StickyFacetBuilder stickyFacetBuilder;
-    if (IssueFilterParameters.FACET_MODE_DEBT.equals(query.facetMode())) {
-      stickyFacetBuilder = new StickyFacetBuilder(esQuery, filters, DEBT_AGGREGATION, DEBT_AGGREGATION_ORDER);
-    } else {
-      stickyFacetBuilder = new StickyFacetBuilder(esQuery, filters);
+  private static StickyFacetBuilder newStickyFacetBuilder(IssueQuery query, Map<String, FilterBuilder> filters, QueryBuilder esQuery) {
+    if (hasQueryEffortFacet(query)) {
+      return new StickyFacetBuilder(esQuery, filters, EFFORT_AGGREGATION, EFFORT_AGGREGATION_ORDER);
     }
-    return stickyFacetBuilder;
+    return new StickyFacetBuilder(esQuery, filters);
   }
 
-  private void addSimpleStickyFacetIfNeeded(SearchOptions options, StickyFacetBuilder stickyFacetBuilder, SearchRequestBuilder esSearch,
+  private static void addSimpleStickyFacetIfNeeded(SearchOptions options, StickyFacetBuilder stickyFacetBuilder, SearchRequestBuilder esSearch,
     String facetName, String fieldName, Object... selectedValues) {
     if (options.getFacets().contains(facetName)) {
       esSearch.addAggregation(stickyFacetBuilder.buildStickyFacet(fieldName, facetName, DEFAULT_FACET_SIZE, selectedValues));
     }
   }
 
-  private AggregationBuilder addDebtAggregationIfNeeded(IssueQuery query, AggregationBuilder aggregation) {
-    if (IssueFilterParameters.FACET_MODE_DEBT.equals(query.facetMode())) {
-      aggregation.subAggregation(DEBT_AGGREGATION);
+  private static AggregationBuilder addEffortAggregationIfNeeded(IssueQuery query, AggregationBuilder aggregation) {
+    if (hasQueryEffortFacet(query)) {
+      aggregation.subAggregation(EFFORT_AGGREGATION);
     }
     return aggregation;
   }
 
+  private static boolean hasQueryEffortFacet(IssueQuery query) {
+    return FACET_MODE_EFFORT.equals(query.facetMode()) || DEPRECATED_FACET_MODE_DEBT.equals(query.facetMode());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "DEPRECATED_FACET_MODE_DEBT"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)